common/dialer/resolver.go

```
@@ -41,6 +52,16 @@ func (r *EnvResolver) Resolve() (string, error) {
 	return os.Getenv(r.key), nil
 }
 
+// ResolveAll retuns the env key in a slice
+func (r *EnvResolver) ResolveAll() ([]string, error) {
+	all := []string{}
+	s, err := r.Resolve()
+	if s != "" {
+		all = append(all, s)
+	}
```

should we return an err if s == ""?

---

Not 100% clear from the interface, but I think it's ok to return an empty result, where empty is "" for Resolve and []string{} for ResolveAll, if the resolution logic was successful and there were just no results. For these simple resolvers that essentially means no errors, but gives room to error in more complicated resolvers e.g. service discovery lookups where an http call can fail.

---

Fair - makes sense.